src/Elastic.Apm/Api/Context.cs

```diff
@@ -1,45 +1,50 @@
-using System;
-using System.Collections.Generic;
-using Elastic.Apm.Report.Serialization;
-using Newtonsoft.Json;
-
-namespace Elastic.Apm.Api
-{
-	public class Context
-	{
+using System;
+using System.Collections.Generic;
+using Elastic.Apm.Report.Serialization;
+using Newtonsoft.Json;
+
+namespace Elastic.Apm.Api
+{
+	public class Context
+	{
+		private readonly Lazy<Dictionary<string, string>> _custom = new Lazy<Dictionary<string, string>>();
 		private readonly Lazy<Dictionary<string, string>> _labels = new Lazy<Dictionary<string, string>>();
 
-		/// <summary>
-		/// <seealso cref="ShouldSerializeLabels" />
-		/// </summary>
-		[JsonProperty("tags")]
-		[JsonConverter(typeof(LabelsJsonConverter))]
+		public Dictionary<string, string> Custom => _custom.Value;
```

Comment:
We need `JsonConverter` here as well to ensure the keys are valid.

Reply:
Added a converter for this - also covered by tests now.